api/src/main/java/org/openmrs/module/registrationcore/api/mpi/common/MpiAuthenticator.java

```diff
@@ -0,0 +1,28 @@
+package org.openmrs.module.registrationcore.api.mpi.common;
+
+/**
+ * This service perform authentication to MPI server. It is a Spring managed bean which is configured
+ * in moduleApplicationContext.xml.
+ */
+public interface MpiAuthenticator {
```

Comment:
@djazayeri actually calling to authenticator is placed in OpenEmpiConnector and that is ok for me. Sure, we can put it for example in RestQueryExecutor.